eslint-rules/lib/rules/prop-value-shape-deprecation.js

```diff
@@ -0,0 +1,141 @@
+const _ = require('lodash');
+const utils = require('../utils');
+
+const MAP_SCHEMA = {
+  type: 'object',
+  properties: {
+    components: {
+      type: 'array',
+      items: {
+        type: 'string'
+      }
+    },
+    propNames: {
+      type: 'array',
+      items: {
+        type: 'string'
+      }
+    },
+    shape: {
+      type: 'array',
+      items: {
+        type: 'object',
+        required: ['prop', 'message', 'fix'],
+        properties: {
+          prop: {
+            type: 'string'
+          },
+          message: {
+            type: 'string'
+          },
+          fix: {
+            type: 'object',
+            additionalProperties: true
+          }
+        }
+      }
+    }
+  },
+  additionalProperties: true
+};
+
+module.exports = {
+  meta: {
+    docs: {
+      description: "Detect deprecation of prop's value shape",
+      category: 'Best Practices',
+      recommended: true
+    },
+    messages: {
+      uiLib: "This prop's value shape is deprecated."
+    },
+    fixable: 'code',
+    schema: [MAP_SCHEMA]
+  },
+  create(context) {
+    function testJSXAttributes(node) {
+      try {
+        const {deprecations} = _.get(context, 'options[0]');
+        const nodeName = _.get(node, 'name.name');
+        _.forEach(deprecations, deprecation => {
```

Comment:
In other rules we check if we have imports that are relevant to this rule.
I'm not sure that our previous way is better since it doesn't seem to take any time to do this on a large example (yes, I've tried on a somewhat large screen :) ).
Well done!

Reply:
I see now that you don't make sure it's the correct import in another way (I assumed you did).
Consider a case where someone is using a component from `react-native` and not `wix-react-native-ui-lib` and it has the same prop name (we have at least a few components with the same name), we will notify a deprecation on a source we don't control.
If you look at `component_deprecation` you'll see `"source": "module-with-deprecations",`, I'm not sure you'll need to add support for that as well, but it might be needed to send in the import name.

Reply:
@lidord-wix, I'm working on fixing a bug in the `component-deprecation`.
I'm basically creating a new rule for component-prop-deprecation (later we will remove this part from the old rule and fix that rule as well).
I am creating new functions in the utils, based on the idea (you started 👍 ) to get all the data from the node (without creating the imports list etc).
You might want to wait for it to be ready before continuing this PR.

Reply:
No problem!
About the import - I consulted with Ethan when I worked on this ticket and we decided not to check the import in this rule, we can consider it again if you think it is important.